src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/HttpDigestFactory.php

```diff
@@ -58,10 +58,26 @@ public function getKey()
     public function addConfiguration(NodeDefinition $node)
     {
         $node
+            ->beforeNormalization()
+                ->ifTrue(function ($v) { return isset($v['key']); })
```

Comment:
``array_key_exist`` may be better, to acount for ``null`` values (otherwise they would not trigger the BC layer and then be reported as invalid keys).
I'm not sure it is necessary for this particular case though, as empty values are not allowed for the secret, and so no existing (working) project can have ``null`` in this setting. But we may need to check other places